source/validate_type_unique.cpp

```diff
@@ -26,9 +26,15 @@ namespace libspirv {
 // Validates that type declarations are unique, unless multiple declarations
 // of the same data type are allowed by the specification.
 // (see section 2.8 Types and Variables)
+// Doesn't do anything if SPV_VAL_ignore_type_decl_unique was declared in the
+// module.
```

Comment:
Ok.  But please add a comment here saying we're adding this only for some limited legacy uses, and this mechanism should not otherwise be used.